### PR TITLE
[Yaml] parse empty sequence elements as null

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Yaml/Tests/InlineTest.php [hunk 1]
```diff
@@ -1004,4 +1004,10 @@ public function testParseQuotedReferenceLikeStringsInSequence()
 
         $this->assertSame(['&foo', '&bar', '&baz'], Inline::parse($yaml));
     }
+
+    public function testParseSequenceWithEmptyElement()
+    {
+        $this->assertSame(['foo', null, 'bar'], Inline::parse('[foo, , bar]'));
+        $this->assertSame([null, 'foo', 'bar'], Inline::parse('[, foo, bar]'));
```
Comment (alexandre-daubois) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php:1011]: Should we add an assertion with `[foo, bar, ]`?

Comment (xabbuh) [reply src/Symfony/Component/Yaml/Tests/InlineTest.php:1011]: as far as I can see `[foo, bar, ]` parses to `['foo', 'bar']` and not `['foo', 'bar', null]`.

Comment (xabbuh) [reply src/Symfony/Component/Yaml/Tests/InlineTest.php:1011]: but we can still add a test for this indeed

Comment (alexandre-daubois) [reply src/Symfony/Component/Yaml/Tests/InlineTest.php:1011]: Yes, it's just to ensure that the last element is not parsed as expected 👍

**Review by alexandre-daubois — APPROVED** [review]
Thank you for taking care of this! It will make things way easier for #58243

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**